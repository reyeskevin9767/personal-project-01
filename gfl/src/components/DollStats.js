import React from 'react';

const DollStats = ({ dollStats, dummyLink }) => {
//* Deconstrcuted all values from dollStats including adding ones specific
//* to certain T-Dolls
  const {
    armor = 0,
    armorPiercing,
    criticalPercent,
    dodge,
    hit,
    hp,
    pow,
    rate,
    speed,
    critDamage = 50,
  } = dollStats;

  return (
    <div>
      <p>Health - {(hp / 5) * dummyLink}</p>
      <p>Damage - {pow}</p>
      <p>Evasion - {dodge}</p>
      <p>Accuracy - {hit}</p>
      <p>Rate of Fire - {rate}</p>
      <p>Movement Speed - {speed}</p>
      <p>Armor - {armor}%</p>
      <p>Crit. Rate - {criticalPercent}%</p>
      <p>Crit. Damage - {critDamage}%</p>
      <p>Armor Piercing - {armorPiercing}</p>
    </div>
  );
};

export default DollStats;
