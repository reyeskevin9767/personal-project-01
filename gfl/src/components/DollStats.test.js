import React from 'react';
import { mount } from 'enzyme';
import renderer from "react-test-renderer";
import DollStats from './DollStats';

//* Testing the DollStats Component
describe('testing DollStats component', () => {
  it('test if screenshot matches', () => {
    const dollStats = {
      armorPiercing: 15,
      criticalPercent: 20,
      dodge: 66,
      hit: 56,
      hp: 380,
      pow: 29,
      rate: 61,
      speed: 15,
    };
    const component = renderer.create(<DollStats dollStats={dollStats} dummyLink={5} />);
    expect(component.toJSON()).toMatchSnapshot();
  });





});
