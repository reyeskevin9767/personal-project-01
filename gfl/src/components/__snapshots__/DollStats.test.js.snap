// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing DollStats component test if screenshot matches 1`] = `
<div>
  <p>
    Health - 
    380
  </p>
  <p>
    Damage - 
    29
  </p>
  <p>
    Evasion - 
    66
  </p>
  <p>
    Accuracy - 
    56
  </p>
  <p>
    Rate of Fire - 
    61
  </p>
  <p>
    Movement Speed - 
    15
  </p>
  <p>
    Armor - 
    0
    %
  </p>
  <p>
    Crit. Rate - 
    20
    %
  </p>
  <p>
    Crit. Damage - 
    50
    %
  </p>
  <p>
    Armor Piercing - 
    15
  </p>
</div>
`;
